vllm/attention/layer.py

```diff
@@ -93,16 +93,28 @@ def forward(
         kv_cache: Optional[torch.Tensor],
         attn_metadata: AttentionMetadata,
         attn_type: AttentionType = AttentionType.DECODER,
+        compute_new_kv: bool = True,
     ) -> torch.Tensor:
 
-        return self.impl.forward(query,
-                                 key,
-                                 value,
-                                 kv_cache,
-                                 attn_metadata,
-                                 self._k_scale,
-                                 self._v_scale,
-                                 attn_type=attn_type)
+        if isinstance(self.impl, XFormersCLAImpl):
+            return self.impl.forward(query,
+                                    key,
+                                    value,
+                                    kv_cache,
+                                    attn_metadata,
+                                    self._k_scale,
+                                    self._v_scale,
+                                    attn_type=attn_type,
+                                    compute_new_kv=compute_new_kv)
```

Comment:
I'm not sure if it is, but wondering if it's possible to deduce this property within XFormersCLAImpl instead of passing it in here? maybe via a property of the kv_cache itself? 

It would be nice to keep the interface identical if it's possible (without other downsides).

Reply:
Made `compute_new_kv` an attribute of `XFormersCLAImpl`.